src/modules/75-cf/components/FlagArchiving/useRestoreFlagDialog.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import { Intent } from '@harness/design-system'
+import { UseConfirmationDialogReturn, useToaster } from '@harness/uicore'
+import { String, useStrings } from 'framework/strings'
+import { useRestoreFeatureFlag, RestoreFeatureFlagQueryParams, Feature } from 'services/cf'
+import useResponseError from '@cf/hooks/useResponseError'
+import { useConfirmAction } from '@common/hooks'
+
+export interface RestoreFlagDialogProps {
+  flagData: Feature
+  queryParams: RestoreFeatureFlagQueryParams
+  refetchFlags?: () => void
+}
+
+const useRestoreFlagDialog = ({
+  flagData,
+  queryParams: restoreQueryParams,
+  refetchFlags
+}: RestoreFlagDialogProps): UseConfirmationDialogReturn['openDialog'] => {
+  const { showSuccess, showError } = useToaster()
+  const { getString } = useStrings()
+  const { handleResponseError } = useResponseError()
+
+  const queryParams = {
+    identifier: flagData.identifier,
+    environmentIdentifier: flagData.envProperties?.environment,
+    ...restoreQueryParams
+  }
+
+  const { mutate: restoreFeatureFlag } = useRestoreFeatureFlag({
+    identifier: flagData.identifier,
+    queryParams
+  })
+
+  const handleRestore = async (): Promise<void> => {
+    try {
+      await restoreFeatureFlag()
+      showSuccess(getString('cf.featureFlags.archiving.restoreSuccess'))
+      refetchFlags?.()
```

Comment:
You can get rid of having to do this by defaulting `refetchFlags` to `() => void 0`. `() => {}` would also work, but ESLint will whinge at you
```suggestion
      refetchFlags()
```